### ISSUE TITLE
TypeScript errors when using 6.0.3-beta.1 and `module: "NodeNext"`

### ISSUE THREAD

**Issue by narthur** [oazapfts/oazapfts#589]
I'm trying to use NodeNext for my typescript module option in [this repo](https://github.com/narthur/baserow-sdk). Based on [this issue](https://github.com/oazapfts/oazapfts/issues/578) I switched to 6.0.3-beta.1 to make it work. However I'm still seeing typescript errors in oazapfts' output.

For example, the first part of the generated code is as follows:

```ts
/**
 * Baserow API spec
 * 1.22.3
 * DO NOT MODIFY - This file has been generated using oazapfts.
 * See https://www.npmjs.com/package/oazapfts
 */
import * as Oazapfts from "@oazapfts/runtime";
import * as QS from "@oazapfts/runtime/query";
export const defaults: Oazapfts.Defaults<Oazapfts.CustomHeaders> = {
    headers: {},
    baseUrl: "/",
};
const oazapfts = Oazapfts.runtime(defaults);
export const servers = {};
export type EmailTesterRequest = {
    target_email: string;
};
```

This code results in the following three typescript errors:

> src/__generated__/baserow.ts(9,33): 
error TS2694: Namespace '"/Users/work/ProgrammingProjects/baserow-sdk/node_modules/.pnpm/\@oazapfts+runtime@1.0.1/node_modules/@oazapfts/runtime/index"' has no exported member 'Defaults'.
> 
> src/__generated__/baserow.ts(9,51): error TS2694: 
Namespace '"/Users/work/ProgrammingProjects/baserow-sdk/node_modules/.pnpm/\@oazapfts+runtime@1.0.1/node_modules/@oazapfts/runtime/index"' has no exported member 'CustomHeaders'.
> 
> src/__generated__/baserow.ts(13,27): error TS2339: 
Property 'runtime' does not exist on type 'typeof import("/Users/work/ProgrammingProjects/baserow-sdk/node_modules/.pnpm/\@oazapfts+runtime@1.0.1/node_modules/@oazapfts/runtime/index", { with: { "resolution-mode": "import" } })'.

Plus many more similar throughout the file.

What do I need to do to get oazapfts working with typescript NodeNext?

tsconfig.json:

```json
{
  "compilerOptions": {
    "target": "es2020",
    "module": "NodeNext",
    "moduleResolution": "NodeNext",
    "declaration": true,
    "outDir": "./dist",
    "declarationDir": "./dist",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true
  }
}
```

package.json:

```json
{
  "name": "baserow-sdk",
  "version": "1.0.0",
  "description": "",
  "main": "dist/index.js",
  "type": "module",
  "files": [
    "dist/**/*.js",
    "dist/**/*.d.ts"
  ],
  "typings": "dist/index.d.ts",
  "scripts": {
    "test": "vitest",
    "prepare": "pnpm generate && tsc",
    "generate": "mkdir -p src/__generated__ && oazapfts ./openapi.json ./src/__generated__/baserow.ts",
    "lint": "eslint .",
    "format": "prettier --write .",
    "checkTs": "tsc --noEmit"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@oazapfts/runtime": "^1.0.1",
    "oazapfts": "6.0.3-beta.1",
    "p-limit": "^5.0.0",
    "typescript": "^5.3.3"
  },
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "^7.0.2",
    "@typescript-eslint/parser": "^7.0.2",
    "eslint": "^8.56.0",
    "prettier": "^3.2.5",
    "vitest": "^1.3.1"
  }
}
```

**Comment by Xiphe** [comment]
I can confirm this. having a look now

**Comment by Xiphe** [comment]
@narthur thanks for raising this 👍

Should be fixed in 6.0.4, let me know when there's anything else.
